src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -525,3 +661,126 @@ CreateTruncateTrigger(Oid relationId)
 	CreateTrigger(trigger, NULL, relationId, InvalidOid, InvalidOid, InvalidOid,
 				  internal);
 }
+
+
+/*
+ * ColocationId searches pg_dist_colocation for shard count, replication factor
+ * and distribution column type. If a matching entry is found, it returns the
+ * colocation id, otherwise it returns INVALID_COLOCATION_ID.
+ */
+static uint32
+ColocationId(int shardCount, int replicationFactor, Oid distributionColumnType)
+{
+	uint32 colocationId = INVALID_COLOCATION_ID;
+	HeapTuple colocationTuple = NULL;
+	SysScanDesc scanDescriptor;
+	ScanKeyData scanKey[3];
+	int scanKeyCount = 3;
```

Comment:
Instead of having duplicate numbers, you could make scanKeyCount a constant and use this constant as the scanKey array size